how-to/integrate-with-rss/client/src/browser.ts

```diff
@@ -14,3 +16,184 @@ export async function launchView(
 	}
 	return platform.createView(viewOptions, targetIdentity, targetView);
 }
+
+export async function launchPage(
+	page: Page,
+	targetWindowIdentity?: OpenFin.Identity
+): Promise<BrowserWindowModule | undefined> {
+	if (targetWindowIdentity) {
+		const window = await findWindow(targetWindowIdentity);
+		if (window) {
+			const allPages = await window.getPages();
+			for (const allPage of allPages) {
+				if (allPage.pageId === page.pageId) {
+					await window.setActivePage(page.pageId);
+					return window;
+				}
+			}
+			await window.addPage(page);
+			await window.setActivePage(page.pageId);
+			return window;
+		}
+	}
+
+	const platform = getCurrentSync();
+	return platform.Browser.createWindow({
+		workspacePlatform: {
+			pages: [page]
+		}
+	});
+}
+
+export async function launchWindow(
+	options: BrowserCreateWindowRequest,
+	reuse?: boolean
+): Promise<BrowserWindowModule | undefined> {
+	const platform = getCurrentSync();
+
+	if (reuse) {
+		const window = await findWindow({ name: options.name, uuid: options.uuid });
+		if (window) {
+			return window;
+		}
+	}
+
+	return platform.Browser.createWindow(options);
+}
+
+export async function findAndActivateView(
+	windowIdentity: OpenFin.Identity,
```

Comment:
A different approach could be:
if(view has name...check to see if it exists).
view = fin.View.wrapSync(identity);

If it does get the parent window. And from there you can find the page etc.

It means you don't need two functions. If you specify the same identity then OpenFin will not create a second instance. You can find it and then focus it.